ducktape/cluster/node_container.py

```diff
@@ -112,16 +117,43 @@ def remove_spec(self, cluster_spec):
         :throws InsufficientResourcesError:     If there are not enough nodes in the NodeContainer.
                                                 Nothing will be removed unless enough are available.
```

Comment:
this kinda seems like a lie, i feel like yes it should throw an insufficient resource error as thats more pythonic, but instead it returns an error?,
also please update the return.

Reply:
yeah, this is a lie :D I'll update the doc